packages/pretzelai-extension/src/migrations/migration_functions.ts

```diff
@@ -0,0 +1,600 @@
+/* eslint-disable camelcase */
+/*
+ * Copyright (c) Pretzel AI GmbH.
+ * This file is part of the Pretzel project and is licensed under the
+ * GNU Affero General Public License version 3.
+ * See the LICENSE_AGPLv3 file at the root of the project for the full license text.
+ * Contributions by contributors listed in the PRETZEL_CONTRIBUTORS file (found at
+ * the root of the project) are licensed under AGPLv3.
+ */
+
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+function addMissingSettings(target, source) {
+  const clonedTarget = structuredClone(target);
+  for (const key in source) {
+    if (Object.prototype.hasOwnProperty.call(source, key)) {
+      if (!(key in clonedTarget)) {
+        clonedTarget[key] = structuredClone(source[key]);
+      } else if (typeof source[key] === 'object' && source[key] !== null) {
+        if (typeof clonedTarget[key] !== 'object' || clonedTarget[key] === null) {
+          clonedTarget[key] = {};
+        }
+        clonedTarget[key] = addMissingSettings(clonedTarget[key], source[key]);
+      }
+    }
+  }
+  return clonedTarget;
+}
+
+function addPropertiesMigration(settings, migration_name) {
+  const newSettings = addMissingSettings(settings, returnDefaults()) as ReturnType<typeof returnDefaults>;
+  newSettings.version = migration_name.split('_to_')[1];
+  return newSettings;
+}
+
+export const migration_functions = {
+  // To just add properties in your migration:
```

Comment:
@prasoon2211 Basically I refactored settings files so these are the steps if you just want to add new properties. Before you had to write a shit-ton of boilerplate like if we are working at Google or something